myenumerables.rb

```diff
@@ -0,0 +1,158 @@
+module Enumerable
+  def my_each
+    index = 0
+    array = to_a
+    if block_given?
+      for item in 0..array.length - 1
+        yield(self[item])
+      end
+    else
+      while index < array.size
+        return to_enum(:my_each) unless block_given?
+
+        yield self[index]
+        index += 1
+        return self
+      end
+    end
+  end
+
+  def my_each_with_index(index = 0)
+    while index < size
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield self[index], index
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result << item if yield(item) == true }
+    result
+  end
+
+  def my_all?(query = nil)
+    if query.class == Class
+      my_each do |idx|
+        return false unless idx.class == query
+      end
+    elsif query.class == String or query.class == Integer
+      my_each do |idx|
+        return false unless idx == query
+      end
+    elsif query.class == Regexp
+      my_each do |idx|
+        return false unless idx.match(query)
+      end
+    elsif block_given?
+      my_each do |idx|
+        return false unless yield(idx)
+      end
+    elsif query.nil?
+      my_each do |idx|
+        return false if idx === false
+      end
+    else
+      my_each do |idx|
+        return false unless !idx.nil?
+      end
+    end
+    true
+  end
+
+  def my_any?(*query)
+    if block_given?
+      length.times { |item| return true unless yield(item) }
+    elsif query.class == Regexp
+      length.times { |item| return true unless item == query }
+    elsif query.class == Class
+      length.times { |item| return true unless item.class == query }
+    elsif query.class == Numeric or query.class == String
+      length.times { |item| return true unless item == query }
+    elsif query.class == Proc
+      length.times { |item| return true unless yield(item) }
+    else
+      length.times { |item| return true unless item }
+    end
+    false
+  end
```

Comment:
There is that error
```rb
2.6.3 :014 > false_block =  proc { |num| num > 9 } 
 => #<Proc:0x00000000024600f8@(irb):14> 
2.6.3 :015 > array.any?(&false_block)
 => false 
2.6.3 :016 > array.my_any?(&false_block)
 => true 
2.6.3 :017 > 
```

Reply:
```rb
2.6.3 :018 > true_array = [nil, false, true, []]
 => [nil, false, true, []] 
2.6.3 :019 > true_array.my_any?
 => false 
2.6.3 :020 > true_array.any?
 => true 
2.6.3 :021 > 
```

Reply:
```rb
2.6.3 :024 > array
 => [3, 8, 5, 6, 1, 5, 8, 3, 7, 7, 6, 0, 3, 6, 7, 7, 2, 2, 1, 1, 4, 3, 5, 1, 4, 5, 2, 4, 8, 7, 2, 7, 8, 2, 2, 7, 4, 0, 6, 5, 7, 7, 6, 4, 1, 8, 5, 5, 7, 0, 3, 3, 1, 0, 1, 5, 5, 7, 3, 8, 5, 2, 1, 5, 3, 8, 6, 3, 5, 5, 6, 4, 0, 7, 5, 7, 7, 7, 7, 8, 1, 6, 8, 3, 1, 6, 5, 6, 5, 7, 4, 1, 8, 8, 6, 1, 1, 0, 8, 5] 
2.6.3 :025 > array.any?(Integer)
 => true 
2.6.3 :026 > array.my_any?(Integer)
 => false 
2.6.3 :027 >
```

Reply:
https://github.com/rootDEV2990/ruby_enumerables_rewrites/pull/1#discussion_r429649955 still persist

Reply:
https://github.com/rootDEV2990/ruby_enumerables_rewrites/pull/1#discussion_r429649808 still persist